internal/operands/common-instancetypes/reconcile.go

```diff
@@ -0,0 +1,147 @@
+	var bundle []C
+	for {
+		resource := new(C)
+		err = decoder.Decode(resource)
+		if err == io.EOF {
+			return bundle, nil
+		}
+		if err != nil {
+			return nil, err
+		}
+		bundle = append(bundle, *resource)
+	}
+}
+
+func New(instancetypeBundlePath, preferenceBundlePath string) (operands.Operand, error) {
+	virtualMachineClusterInstancetypes, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterInstancetype](instancetypeBundlePath)
+	if err != nil {
+		return nil, err
+	}
+	virtualMachineClusterPreferences, err := fetchClusterResources[instancetypev1alpha2.VirtualMachineClusterPreference](preferenceBundlePath)
+	if err != nil {
+		return nil, err
+	}
+	return &commonInstancetypes{
+		virtualMachineClusterInstancetypes: virtualMachineClusterInstancetypes,
+		virtualMachineClusterPreferences:   virtualMachineClusterPreferences,
+	}, nil
+}
+
+func (c *commonInstancetypes) Name() string {
+	return operandName
+}
+
+func (c *commonInstancetypes) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &instancetypev1alpha2.VirtualMachineClusterInstancetype{}, Crd: instancetypeapi.ClusterPluralResourceName, WatchFullObject: true},
+		{Object: &instancetypev1alpha2.VirtualMachineClusterPreference{}, Crd: instancetypeapi.ClusterPluralPreferenceResourceName, WatchFullObject: true},
+	}
+}
+
+func (c *commonInstancetypes) WatchTypes() []operands.WatchType {
+	return nil
+}
+
+func (c *commonInstancetypes) RequiredCrds() []string {
+	return nil
+}
+
+func (c *commonInstancetypes) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	request.Logger.Info("Reconciling common-instancetypes")
+	return common.CollectResourceStatus(request, c.reconcileFuncs()...)
+}
+
+func (c *commonInstancetypes) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	return nil, nil
+}
+
+func (c *commonInstancetypes) reconcileFuncs() []common.ReconcileFunc {
+	funcs := []common.ReconcileFunc{}
+	funcs = append(funcs, c.reconcileVirtualMachineClusterInstancetypesFuncs()...)
+	funcs = append(funcs, c.reconcileVirtualMachineClusterPreferencesFuncs()...)
+	return funcs
+}
+
+func (c *commonInstancetypes) reconcileVirtualMachineClusterInstancetypesFuncs() []common.ReconcileFunc {
+	funcs := make([]common.ReconcileFunc, 0, len(c.virtualMachineClusterInstancetypes))
+	for i := range c.virtualMachineClusterInstancetypes {
+		clusterInstancetype := &c.virtualMachineClusterInstancetypes[i]
+		funcs = append(funcs, func(request *common.Request) (common.ReconcileResult, error) {
+			return common.CreateOrUpdate(request).
+				ClusterResource(clusterInstancetype).
+				WithAppLabels(operandName, operandComponent).
+				UpdateFunc(func(newRes, foundRes client.Object) {
+					foundRes.(*instancetypev1alpha2.VirtualMachineClusterInstancetype).Spec = newRes.(*instancetypev1alpha2.VirtualMachineClusterInstancetype).Spec
+				}).
+				ImmutableSpec(func(resource client.Object) interface{} {
```

Comment:
The `VirtualMachineClusterInstancetype` is not immutable, so this method should not be called.